streams/src/test/java/org/apache/kafka/streams/state/internals/ListValueStoreTest.java

```diff
@@ -0,0 +1,174 @@
+ */
+package org.apache.kafka.streams.state.internals;
+
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.processor.StateStoreContext;
+import org.apache.kafka.streams.processor.internals.MockStreamsMetrics;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.test.InternalMockProcessorContext;
+import org.apache.kafka.test.MockRecordCollector;
+import org.apache.kafka.test.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+
+import static java.util.Arrays.asList;
+import static org.apache.kafka.test.StreamsTestUtils.toList;
+import static org.junit.Assert.assertEquals;
+
+public class ListValueStoreTest {
+    private static final String STORE_NAME = "rocksDB list value store";
+
+    MockRecordCollector recordCollector;
+    KeyValueStore<Integer, String> listStore;
+    InternalMockProcessorContext<Integer, String> context;
+
+    final File baseDir = TestUtils.tempDirectory("test");
+
+    @Before
+    public void setup() {
+        listStore = buildStore(Serdes.Integer(), Serdes.String());
+
+        recordCollector = new MockRecordCollector();
+        context = new InternalMockProcessorContext<>(
+            baseDir,
+            Serdes.String(),
+            Serdes.Integer(),
+            recordCollector,
+            new ThreadCache(
+                new LogContext("testCache"),
+                0,
+                new MockStreamsMetrics(new Metrics())));
+        context.setTime(1L);
+
+        listStore.init((StateStoreContext) context, listStore);
+    }
+
+    @After
+    public void after() {
+        listStore.close();
+    }
+
+    <K, V> KeyValueStore<K, V> buildStore(final Serde<K> keySerde,
+                                          final Serde<V> valueSerde) {
+        return new ListValueStoreBuilder<>(
+            new RocksDbKeyValueBytesStoreSupplier(STORE_NAME, false),
+            keySerde,
+            valueSerde,
+            Time.SYSTEM)
+            .build();
+    }
+
+    @Test
+    public void shouldGetAll() {
+        listStore.put(0, "zero");
+        // should retain duplicates
+        listStore.put(0, "zero again");
+        listStore.put(1, "one");
+        listStore.put(2, "two");
+
+        final KeyValue<Integer, String> zero = KeyValue.pair(0, "zero");
+        final KeyValue<Integer, String> zeroAgain = KeyValue.pair(0, "zero again");
+        final KeyValue<Integer, String> one = KeyValue.pair(1, "one");
+        final KeyValue<Integer, String> two = KeyValue.pair(2, "two");
+
+        assertEquals(
+            asList(zero, zeroAgain, one, two),
+            toList(listStore.all())
```

Comment:
Potential memory leak? `toList` does not close the iterator -- should we update `toList` ? Or make sure we close the iterator in the using try-with-resources?